src/ba-transport.c

```diff
@@ -1638,8 +1652,12 @@ void ba_transport_thread_cleanup(struct ba_transport_thread *th) {
 	ba_transport_thread_bt_release(th);
 
 	/* If we are closing master thread, release underlying BT transport. */
-	if (th->master)
+	if (th->master) {
 		ba_transport_release(t);
+		/* SCO encoder thread on HF cannot transfer audio when BT socket is closed */
+		if (t->profile & BA_TRANSPORT_PROFILE_MASK_HF)
+			ba_transport_thread_set_state(&t->thread_enc, BA_TRANSPORT_THREAD_STATE_NONE);
```

Comment:
Do you have any test case where this is required?

Currently I'm trying to "fix" IO threads start and termination in case of errors and this patch seems to be related :D For full duplex codecs (e.g. SCO, A2DP FastStream) we need both IO threads running, if one fails other should terminated and PCM client notified.

I will go back to this PR after IO threads fix, because maybe it will be more feasible to change property type from boolean to state/enum after all my changes.

Reply:
> Do you have any test case where this is required?

No, it appears to be a relic of some idea I was experimenting with, that was not cleaned up before I submitted the PR. As far as I can tell now, it is not making any difference to the propertieschanged signals.

There is an "annoyance" with HF/HS nodes that the sink PCM remains with "Running true" when the transport is released by the AG, then on the next acquisition it emits "Running false" followed by "Running true". This makes this signal difficult to use for starting/stopping clients on the HF/HS. I think I was trying to persuade bluealsa to emit a property changed signal even though the encoder thread is still running, but never achieved that successfully.